src/lib/create-utils.js

```diff
@@ -0,0 +1,129 @@
+const inquirer = require('inquirer');
+const actorTemplates = require('@apify/actor-templates');
+const { validateActorName } = require('./utils');
+
+const PROGRAMMING_LANGUAGES = ['JavaScript', 'TypeScript', 'Python'];
+
+/**
+ * @param {string} maybeActorName
+ * @returns {Promise<string>}
+ */
+exports.ensureValidActorName = async (maybeActorName) => {
+    if (maybeActorName) {
+        validateActorName(maybeActorName);
+        return maybeActorName;
+    }
+    return promptActorName();
+};
+
+/**
+ * @param {string} maybeTemplateName
+ * @returns {Promise<object>}
+ */
+exports.getTemplateDefinition = async (maybeTemplateName) => {
+    const manifest = await actorTemplates.fetchManifest();
+    if (maybeTemplateName) {
+        return manifest.templates.find((t) => t.name === maybeTemplateName);
+    }
+
+    const programmingLanguage = await promptProgrammingLanguage();
+    while (true) {
+        const templateDefinition = await promptTemplateDefinition(manifest, programmingLanguage);
+        const shouldInstall = await promptTemplateInstallation(templateDefinition);
+        if (shouldInstall) return templateDefinition;
+    }
+};
+
+/**
+ * @returns {Promise<string>}
+ */
+async function promptActorName() {
+    const answer = await inquirer.prompt([{
+        name: 'actorName',
+        message: 'Name of your new actor:',
+        type: 'input',
+        validate: (promptText) => {
+            try {
+                validateActorName(promptText);
+            } catch (err) {
+                return err.message;
+            }
+            return true;
+        },
+    }]);
+    return answer.actorName;
+}
+
+/**
+ * @returns {Promise<string>}
+ */
+async function promptProgrammingLanguage() {
+    const choices = PROGRAMMING_LANGUAGES.map((lang) => ({
+        name: `👉 ${lang}`,
+        value: lang,
+    }));
+    const answer = await inquirer.prompt([{
+        type: 'list',
+        name: 'programmingLanguage',
+        message: 'Choose the programming language of your new actor:',
+        default: choices[0],
+        choices,
+        loop: false,
+    }]);
+    return answer.programmingLanguage;
+}
+
+/**
+ * @param {object} manifest
+ * @param {string} programmingLanguage
+ * @returns {Promise<object>} template definition
+ */
+async function promptTemplateDefinition(manifest, programmingLanguage) {
+    const choices = manifest.templates
+        .filter((t) => {
+            return t.category.toLowerCase() === programmingLanguage.toLowerCase();
```

Comment:
With this, it may easily happen that somebody will mistype the category, and the template gets invisible. 

So I'd add a test here https://github.com/apify/actor-templates/blob/master/test/templates.test.js that the category is always one of those 3.

Plus, I'd interlink the `const PROGRAMMING_LANGUAGES ` here and that test there with some comment.

Reply:
Good idea. I'll probably skip the interlinking though. Like the 28 days, it's probably not worth it 😅

Or you meant "adding a comment" as the interlink?

Reply:
I just meant to leave the comment at both places

Reply:
Can't we extract the categories straight from the manifest? Either directly from the templates, or we could add another field to the manifest with the categories.

That way, if we ever add a new category, even old versions of the CLI will be able to use them.

Plus we should add a unit test to the templates that each template has a category (maybe it's already there, I'm not sure).

Reply:
I think that's more of a design than engineering question. We can do both. I just felt that CLI should control what it displays. So if someone makes a mistake in templates, which is an external API, it won't show weird stuff in the CLI. Like Java**S**cript and Java**s**cript language next to each other. But we can rely on the templates and just display everything as well.

Reply:
Maybe there could be a unit test in the templates that the category is exactly one of the three allowed ones, to prevent those mistakes.

Or maybe we could have some optimistic formatting in the CLI, e.g. `javascript` -> `JavaScript`.

I agree that the CLI should control what it displays, but recently we had a lot of bug reports from people because they were using an old version of the CLI, so we should try to make things forward-compatible as much as possible, because getting people to upgrade is a hard task.

Reply:
In the end I decided that at least for now, **I'll keep the languages hardcoded in the CLI** and will add a test to the templates repo. There are a lot of changes going on now with the templates, so let's revisit this in the future if needed. In the CLI I imagine the first step to be language selection, not category selection, so if at some point in the future someone decides to add "Automation" category to the templates, it kinda breaks the idea and I don't want this to happen automatically.

IMO "Category" should be renamed "Programming language" in the templates and then Technologies are actually the "categories". But I don't wanna do this change now when NUX team is working on the front-end and break things for them.